wyzesense2mqtt.py

```diff
@@ -12,169 +10,266 @@
 import socket
 import sys
 import subprocess
+import time
+import yaml
 
 import paho.mqtt.client as mqtt
+import wyzesense
 from retrying import retry
-from wyzesense_custom import *
-
-# Read configuration file
-CONFIG_FILE = "config/config.json"
-def read_config():
-    with open(CONFIG_FILE) as config_file:
-        data = json.load(config_file)
-    return data
-config = read_config()
-
-# Set Config File Variables
-MQTT_HOST = config['mqtt']['host']
-MQTT_PORT = config['mqtt']['port']
-MQTT_USERNAME = config['mqtt']['username']
-MQTT_PASSWORD = config['mqtt']['password']
-MQTT_CLIENT_ID = config['mqtt']['client_id']
-MQTT_CLEAN_SESSION = config['mqtt']['clean_session']
-MQTT_KEEPALIVE = config['mqtt']['keepalive']
-MQTT_QOS = config['mqtt']['qos']
-MQTT_RETAIN = config['mqtt']['retain']
-PERFORM_HASS_DISCOVERY = config['perform_hass_discovery']
-HASS_TOPIC_ROOT = config['hass_topic_root']
-WYZESENSE2MQTT_TOPIC_ROOT = config['wyzesense2mqtt_topic_root']
-USB_DEVICE = config['usb_device']
-LOG_FILENAME = config['log']['filename']
-LOG_PATH = config['log']['path']
-LOGGING = config['logging']
-
-# Set MQTT Topics
-SCAN_TOPIC = "{0}scan".format(WYZESENSE2MQTT_TOPIC_ROOT)
-SCAN_RESULT_TOPIC = "{0}scan_result".format(WYZESENSE2MQTT_TOPIC_ROOT)
-REMOVE_TOPIC = "{0}remove".format(WYZESENSE2MQTT_TOPIC_ROOT)
-
-diff = lambda l1, l2: [x for x in l1 if x not in l2]
 
+# Configuration File Locations
+LOGGING_CONFIG_FILE = "config/logging.yaml"
+GENERAL_CONFIG_FILE = "config/config.yaml"
+SENSORS_CONFIG_FILE = "config/sensors.yaml"
+
+# Read data from YAML file
+def read_yaml_file(filename):
+    try:
+        with open(filename) as yaml_file:
+            data = yaml.safe_load(yaml_file)
+            return data
+    except IOError as error:
+        if LOGGER is None:
+            print("File error: {0}".format(str(error)))
```

Comment:
I am, and I will. This will likely help with some of the overly long lines that flake8 complained about and may allow resimplification of those lines.

Reply:
Converted to f-string format in recent commit.